File: Doomsday/Go/doomsday.go
package main

import (
	"fmt"
	"log"
	"strconv"
)

func main() {

	weekdays := [7]string{"Sunday",
		"Monday",
		"Tuesday",
		"Wednesday",
		"Thursday",
		"Friday",
		"Saturday",
	}
	const weeklength = 7
	const century = 100
	const year_cycle = 12
	const century_cycle = 4
	var year string
	fmt.Printf("Enter a year to find its doomsday: ")
	_, err := fmt.Scanln(&year)
	if err != nil {
		log.Fatal(err)
	}
	intyear, err := strconv.ParseInt(year, 0, 0)
  if err != nil {
    log.Fatal(err)
  }

  high_years := intyear/century
  r := high_years % century_cycle
  var anchor int64
  switch r {
  // Tuesday
  case 0:
    anchor = 2
  // Sunday
  case 1:
    anchor = 0
  // Friday
  case 2:
    anchor = 5
  // Wednesday
  case 3:
    anchor = 3
  }

  low_years := intyear % century
  a := low_years / year_cycle
  b := low_years % year_cycle
  c := b / century_cycle
  d := ( a + b + c ) % weeklength
  doomsday := ( anchor + d ) % weeklength
  fmt.Println(weekdays[doomsday])


}

File: algorithms/math/doomsday/cpp/doomsday.cpp
#include <string>

std::string day_of_week(int year, int month, int day) {
    static const int offsets[] = {0, 3, 2, 5, 0, 3, 5, 1, 4, 6, 2, 4};
    static const char* names[] = {
        "Sunday",
        "Monday",
        "Tuesday",
        "Wednesday",
        "Thursday",
        "Friday",
        "Saturday",
    };
    year -= month < 3 ? 1 : 0;
    int index = (year + year / 4 - year / 100 + year / 400 + offsets[month - 1] + day) % 7;
    return names[index];
}
